### ISSUE TITLE
Intersection over Union decreases with increased P+R

### ISSUE THREAD

**Issue by flixmk** [ultralytics/yolov5#1732]
## ❔Question
Hi, 
I am training a model to detect a single type of cell in an 416x416 image. How can i interpret metrics like precision and recall compared to the IoU?

## Additional context
I made a first training attempt with the following results:
![Download](https://user-images.githubusercontent.com/66331106/102622049-c8f2cd80-4140-11eb-97d1-755c221df672.png)

Number of training examples: 27k (augmented)

Then i tried to double the trainingdata to ~53k (augmented) :
![results](https://user-images.githubusercontent.com/66331106/102622199-fa6b9900-4140-11eb-8d5e-f4bca881a5e3.png)

The metrics look a lot better! But when seeing the results for my self on predicted cells, there is no significant improvement. The second model with more data detects some cells that were not detected previously but also vice versa. On top of that I try to calculate the mean IoU. I compare every bounding box of the ground truth with all of the predicted ones in one image. And if the intersection is 0 I wont count it. (Of course there are still some overlaps that are not meant to be included but since i test on the same test data it should not make a difference)
The first model is quite a bit better in my calculations. 

Also the second model tends to detect way more cells in the training data then in the test data compared to the first model. Might this be a sign of overfitting?

Another question that might be lead to something: if i split my dataset int train,val,test then are the metrics calculated on train and val right? So those metrics dont say anything about the performance on my test set right?

Does someone know why i get those results or can someone help me interpret what is going on?

Thanks in advance :D

**Comment by glenn-jocher** [comment]
@Kraufel more training data will always produce better results. I can't comment on your own analysis methods.

**Comment by flixmk** [comment]
@glenn-jocher Yes the metrics show that the results should be significantly better. But when trying the inference on a bunch of images i cant tell the difference. Both models detect a bunch of cells (some cells get detected by both models, some only by one), neither detects all. 

And does the model test against the test set aswell?

**Comment by glenn-jocher** [comment]
@Kraufel the model is _objectively_ better as the metrics show. Your _subjective_ interpretation is another matter and up to you. 

My job is to produce performant models on objective metrics, not opinions on anecdotal results.

**Comment by flixmk** [comment]
@glenn-jocher Thank you  :D I know that it is objectively better. I just wondered how it comes that they are detecting different cells on the same image, when trained on the same data (the larger dataset ist just flipped). Thought there might be more to it.

If not then I will try train it for longer.

**Comment by glenn-jocher** [comment]
@Kraufel train with default settings for best results.
